lib/licenses/helpers.nix

```diff
@@ -0,0 +1,133 @@
+{ lib }:
+let
+  recCheck =
+    check: attrs:
+    if attrs.type == "simple" then
+      check attrs
+    else if attrs.type == "compound" then
+      if attrs.operator == "OR" then
+        recCheck check attrs.base || recCheck check attrs.mod
+      else if attrs.operator == "AND" then
+        recCheck check attrs.base && recCheck check attrs.mod
+      else if attrs.operator == "WITH" then
+        recCheck check attrs.base && recCheck check attrs.mod
+      else
+        throw "Unknown license operator"
+    else
+      throw "Unknown license type or legacy license";
+  recCheck' =
+    check: attrs:
+    if attrs.type == "simple" then
+      check attrs
+    else if attrs.type == "compound" then
+      if attrs.operator == "OR" then
+        recCheck check attrs.base && recCheck check attrs.mod
+      else if attrs.operator == "AND" then
+        recCheck check attrs.base || recCheck check attrs.mod
+      else if attrs.operator == "WITH" then
+        recCheck check attrs.base || recCheck check attrs.mod
+      else
+        throw "Unknown license operator"
+    else
+      throw "Unknown license type or legacy license";
+in
+rec {
+  /**
+    Check wether a license is unfree.
+
+    # Example
+
+    ```nix
+    isUnfree lib.licenses.mit
+    => true
+    ```
+
+    # Type
+
+    ```
+    isUnfree :: AttrSet -> Bool
+    ```
+
+    # Arguments
+
+    - [license] License to check if unfree
+  */
+  isUnfree = recCheck' (x: !x.free);
+
+  /**
+    Check wether a license is redistributable.
+
+    # Example
+
+    ```nix
+    isRedistributable lib.licenses.mit
```

Comment:
Given that these functions are capable of dealing with composite license expressions, it'd make sense to have examples for those too. (Applies to all of these helpers.)